packages/contracts/contracts/L1/L2OutputOracle.sol

```diff
@@ -85,17 +86,17 @@ contract L2OutputOracle is Initializable, Semver {
      * @param _l2BlockTime               The time per L2 block, in seconds.
      * @param _startingBlockNumber       The number of the first L2 block.
      * @param _startingTimestamp         The timestamp of the first L2 block.
-     * @param _validator                 The address of the validator.
-     * @param _challenger                The address of the challenger.
+     * @param _validatorPool             The address of the ValidatorPool contract.
+     * @param _colosseum                 The address of the Colosseum contract.
      * @param _finalizationPeriodSeconds Output finalization time in seconds.
      */
     constructor(
         uint256 _submissionInterval,
         uint256 _l2BlockTime,
         uint256 _startingBlockNumber,
         uint256 _startingTimestamp,
-        address _validator,
-        address _challenger,
+        ValidatorPool _validatorPool,
+        address _colosseum,
```

Comment:
Circular dependency happens here due to the reference to the `L2OutputOracle` address in the constructor of the `Colosseum` contract. You should pass the `address` to `initialize()` or in other relevant functions. @0xHansLee

Reply:
This is not possible because immutable variables can only be assigned in the constructor.